<a href="https://colab.research.google.com/github/Murdocksilva/Data-Science---Coder-House/blob/main/Preentrega1_Coderhouse_AlfonsoMarrero.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Preentrega 1
#### Data Science I - Coderhouse




## Modelo para predecir la aprobación de créditos  bancarios



### Integrantes:
Alfonso Marrero Silva

### Origen del dataset: https://www.kaggle.com/datasets/architsharma01/loan-approval-prediction-dataset

## Dependencias:


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
from matplotlib.ticker import FormatStrFormatter
from seaborn import boxplot, distplot

In [2]:
pd.set_option('display.max_columns', None)
pd.set_option("display.max_colwidth", None)
sns.set_style('darkgrid')

#Data Acquisition

In [3]:
from google.colab import drive
drive.mount ('/content/drive')

Mounted at /content/drive


In [4]:
df= pd.read_csv('/content/drive/MyDrive/DataScience - Coderhouse/DATA SCIENCE I/DATA SET/loan_approval_dataset.csv')

# Descripción de variables



*   no_of_dependents = número de dependientes del solicitante. >=0
*   education = nivel educativo del solicitante [Graduate : graduado , Not Graduate : no graduado]
*   self_employed = trabajador por su propia cuenta [Yes : independiente , No : no independiente]
*   income_annum = ingreso anualizado. >0
*   loan_amount = monto del préstamo. >0
*   loan_term = meses plazo del préstamo. >0
*   cibil_score = puntaje de solvencia crediticia de los ultimos 6 meses. Mayor puntaje implica mayor solvencia y responsabilidad del solicitante [300,900]
*   residential_assets_value = valor de los activos residenciales.
*   commercial_assets_value = valor de los activos comerciales.
*   luxury_assets_value = valor de los activos de lujo.
*   bank_asset_value = valor de los activos bancarios.
*   loan_status = estado del préstamo. variable target a predecir [Approved : aprobado , Rejected : rechazado]















# Estadística descriptiva

In [5]:
df.shape

(4269, 13)

El dataset obtenido de la base de datos posee 4269 filas y 13 columnas, de las cuales 12 (doce) son variables independientes a analizar y 1 (una) corresponde a la variable target o dependiente

In [6]:
df.columns

Index(['loan_id', ' no_of_dependents', ' education', ' self_employed',
       ' income_annum', ' loan_amount', ' loan_term', ' cibil_score',
       ' residential_assets_value', ' commercial_assets_value',
       ' luxury_assets_value', ' bank_asset_value', ' loan_status'],
      dtype='object')

In [7]:
df.head(10)

loan_id   no_of_dependents      education  self_employed   income_annum  \
0        1                  2       Graduate             No        9600000   
1        2                  0   Not Graduate            Yes        4100000   
2        3                  3       Graduate             No        9100000   
3        4                  3       Graduate             No        8200000   
4        5                  5   Not Graduate            Yes        9800000   
5        6                  0       Graduate            Yes        4800000   
6        7                  5       Graduate             No        8700000   
7        8                  2       Graduate            Yes        5700000   
8        9                  0       Graduate            Yes         800000   
9       10                  5   Not Graduate             No        1100000   

    loan_amount   loan_term   cibil_score   residential_assets_value  \
0      29900000          12           778                    2400000   
1      12200000           8           417                    2700000   
2      29700000          20           506                    7100000   
3      30700000           8           467                   18200000   
4      24200000          20           382                   12400000   
5      13500000          10           319                    6800000   
6      33000000           4           678                   22500000   
7      15000000          20           382                   13200000   
8       2200000          20           782                    1300000   
9       4300000          10           388                    3200000   

    commercial_assets_value   luxury_assets_value   bank_asset_value  \
0                  17600000              22700000            8000000   
1                   2200000               8800000            3300000   
2                   4500000              33300000           12800000   
3                   3300000              23300000            7900000   
4                   8200000              29400000            5000000   
5                   8300000              13700000            5100000   
6                  14800000              29200000            4300000   
7                   5700000              11800000            6000000   
8                    800000               2800000             600000   
9                   1400000               3300000            1600000   

   loan_status  
0     Approved  
1     Rejected  
2     Rejected  
3     Rejected  
4     Rejected  
5     Rejected  
6     Approved  
7     Rejected  
8     Approved  
9     Rejected

In [8]:
df.duplicated().value_counts()

False    4269
Name: count, dtype: int64

El dataset no posee filas repetidas.

In [9]:
df.isna() .sum()

loan_id                      0
 no_of_dependents            0
 education                   0
 self_employed               0
 income_annum                0
 loan_amount                 0
 loan_term                   0
 cibil_score                 0
 residential_assets_value    0
 commercial_assets_value     0
 luxury_assets_value         0
 bank_asset_value            0
 loan_status                 0
dtype: int64

In [10]:
df.info(verbose = True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4269 entries, 0 to 4268
Data columns (total 13 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   loan_id                    4269 non-null   int64 
 1    no_of_dependents          4269 non-null   int64 
 2    education                 4269 non-null   object
 3    self_employed             4269 non-null   object
 4    income_annum              4269 non-null   int64 
 5    loan_amount               4269 non-null   int64 
 6    loan_term                 4269 non-null   int64 
 7    cibil_score               4269 non-null   int64 
 8    residential_assets_value  4269 non-null   int64 
 9    commercial_assets_value   4269 non-null   int64 
 10   luxury_assets_value       4269 non-null   int64 
 11   bank_asset_value          4269 non-null   int64 
 12   loan_status               4269 non-null   object
dtypes: int64(10), object(3)
memory usage: 433.7+ KB


El dataset no posee valores nulos.

In [11]:
df.describe(include = 'object').T

count unique        top  freq
 education      4269      2   Graduate  2144
 self_employed  4269      2        Yes  2150
 loan_status    4269      2   Approved  2656

In [12]:
df.describe().T

count          mean           std       min  \
loan_id                    4269.0  2.135000e+03  1.232498e+03       1.0   
 no_of_dependents          4269.0  2.498712e+00  1.695910e+00       0.0   
 income_annum              4269.0  5.059124e+06  2.806840e+06  200000.0   
 loan_amount               4269.0  1.513345e+07  9.043363e+06  300000.0   
 loan_term                 4269.0  1.090045e+01  5.709187e+00       2.0   
 cibil_score               4269.0  5.999361e+02  1.724304e+02     300.0   
 residential_assets_value  4269.0  7.472617e+06  6.503637e+06 -100000.0   
 commercial_assets_value   4269.0  4.973155e+06  4.388966e+06       0.0   
 luxury_assets_value       4269.0  1.512631e+07  9.103754e+06  300000.0   
 bank_asset_value          4269.0  4.976692e+06  3.250185e+06       0.0   

                                 25%         50%         75%         max  
loan_id                       1068.0      2135.0      3202.0      4269.0  
 no_of_dependents                1.0         3.0         4.0         5.0  
 income_annum              2700000.0   5100000.0   7500000.0   9900000.0  
 loan_amount               7700000.0  14500000.0  21500000.0  39500000.0  
 loan_term                       6.0        10.0        16.0        20.0  
 cibil_score                   453.0       600.0       748.0       900.0  
 residential_assets_value  2200000.0   5600000.0  11300000.0  29100000.0  
 commercial_assets_value   1300000.0   3700000.0   7600000.0  19400000.0  
 luxury_assets_value       7500000.0  14600000.0  21700000.0  39200000.0  
 bank_asset_value          2300000.0   4600000.0   7100000.0  14700000.0